docs/index-server.md

```diff
@@ -0,0 +1,216 @@
+earlier being handled by those would be reassigned to some other
+executor using the distribution logic.
+
+**Note:** Cache loading would be done again in the new executor for the
+current query.
+
+## MetaCache DDL
+The show/drop metacache DDL have been modified to operate on the
+executor side cache as well. So when the used fires show cache a new
+column called cache location will indicate whether the cache is from
+executor or driver. For drop cache the user has to enable/disable the
+index server using the dynamic configuration to clear the cache of the
+desired location.
+
+## Fallback
+In case of any failure the index server would fallback to embedded mode
+which means that the JDBCServer would take care of distributed pruning.
+A similar job would be fired by the JDBCServer which would take care of
+pruning using its own executors. If for any reason the embedded mode
+also fails to prune the datamaps then the job would be passed on to
+driver.
+
+**NOTE:** In case of embedded mode a job would be fired to clear the
+cache as data cached in JDBCServer executors would be of no use.
+
+
+## Configurations
+
+##### carbon.properties(JDBCServer) 
+
+| Name     |      Default Value    |  Description |
+|:----------:|:-------------:|:------:       |
+| carbon.enable.index.server       |  false | Enable the use of index server for pruning for the
+whole application.       |
+| carbon.index.server.ip |    NA   |   Specify the IP/HOST on which the server would be started. Better to specify the private IP. | 
+| carbon.index.server.port | NA | The port on which the index server has to be started. |
+| carbon.disable.index.server.fallback | false | Whether to enable/disable fallback for index server
+. Should be used for testing purposes only. |
+|carbon.index.server.max.worker.threads| 500 | Number of RPC handlers to open for accepting the
+requests from JDBC driver. Max accepted value is Integer.Max. |
+|carbon.index.server.max.jobname.length|NA|The max length of the job to show in the index server application UI. For bigger queries this may impact performance as the whole string would be sent from JDBCServer to IndexServer.|
+
+
+##### carbon.properties(IndexServer) 
+
+| Name     |      Default Value    |  Description |
+|:----------:|:-------------:|:------:       |
+| carbon.enable.index.server       |  false | Enable the use of index server for pruning for the
+whole application.       |
+| carbon.index.server.ip |    NA   |   Specify the IP/HOST on which the server would be started. Better to specify the private IP. | 
+| carbon.index.server.port | NA | The port on which the index server has to be started. |
+|carbon.index.server.max.worker.threads| 500 | Number of RPC handlers to open for accepting the
+requests from JDBC driver. Max accepted value is Integer.Max. |
+|carbon.max.executor.lru.cache.size|  NA | Used to specify the max size for executor LRU cache. Mandatory to set fo the user. |
```

Comment:
what is the relation between this parameter and executor configuration when launching by spark-submit? Is it possible to automatically set it by some calculation based on executor configuration, such that user has no need to consider how to set both.

And :
set fo ?

Reply:
@kevinjmh Currently we dont have any calculation to set automatically, need to test and find the optimal value so that it can be done automatically.